main.cpp

```diff
@@ -65,19 +65,59 @@ struct MbedClientDevice device = {
 // Instantiate the class which implements LWM2M Client API (from simpleclient.h)
 MbedClient mbed_client(device);
 
+// Set up a button interrupt for user interaction
+#ifdef MBED_CONF_APP_BUTTON1
+	InterruptIn counter_btn(MBED_CONF_APP_BUTTON1);
+#endif
 
-// In case of K64F board , there is button resource available
-// to change resource value and unregister
-#ifdef TARGET_K64F
-// Set up Hardware interrupt button.
-InterruptIn obs_button(SW2);
-InterruptIn unreg_button(SW3);
-#else
-//In non K64F boards , set up a timer to simulate updating resource,
-// there is no functionality to unregister.
-Ticker timer;
+
+/**
+ * User interaction handler / simulator. Sets up physical button handler and a ticker
+ * for regular updates for the resources.
+ *
+ * MBED_CONF_APP_BUTTON1 is mapped to actual button pin the mbed_app.json file, where you need to
+ * specify board-specific values or leave them undefined if the board does not have buttons.
+ */
+class InteractionProvider {
+
+public:
+	InteractionProvider(Semaphore& updates_sem) : updates(updates_sem) {
+
+		// Set up handler function for the interaction button, if available
+
+#ifdef MBED_CONF_APP_BUTTON1
+		counter_btn.fall(this, &InteractionProvider::counter_button_handler);
 #endif
 
+	    // Use the counter button handler to send an update of endpoint resource values
+		// to connector every 15 seconds periodically.
+	    timer.attach(this, &InteractionProvider::counter_button_handler, 15.0);
```

Comment:
If we do it this way, with boards having button -> the counter will be updated WITH counter + timer.
I was thinking of a separate counter resource and separate timer resource (lwm2m resource).